content_harvester/s3_cache.py

```diff
@@ -0,0 +1,100 @@
+import json
+import boto3
+from typing import Optional
+
+class S3Cache(object):
+    """
+    Create a dictionary-like interface to an s3 bucket
+
+    Parameters:
+        bucket_name (str - required): The name of the s3 bucket
+        prefix (str - optional): A string to prepend to all keys
+            include any necessary slashed when specifying the prefix
+        **kwargs: Passed directly to the boto3 client
+    """
+    def __init__(self, bucket_name: str, prefix: Optional[str]=None, **kwargs):
+        self.bucket_name = bucket_name
+        self.prefix = prefix or ''
+        try:
+            self.s3 = boto3.client('s3', **kwargs)
+            self.s3.head_bucket(Bucket=self.bucket_name)
+        except Exception as e:
+            print(f"Error connecting to s3 cache backend {bucket_name}: {e}")
+            raise e
+    
+    def __getitem__(self, key):
+        """
+        Get the value for the key using bracket notation: S3Cache[key]
+        Raise all exceptions with added key context
+        """
+        try:
+            response = self.s3.get_object(
+                Bucket=self.bucket_name, Key=self.prefix+key)
```

Comment:
Should we wrap this in an f-string just to be safe?

Reply:
Done!